### PR TITLE
Bring EMP grenades in line with reality

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
SUMMARY: Balance "Revamp EMP grenade recipe"

#### Purpose of change
EMP grenades as they currently exist do not exist in real life.
This PR aims to make work as a plausible EMP device would, based off of [this paper](https://apps.dtic.mil/dtic/tr/fulltext/u2/a332511.pdf), but slightly miniaturized from the size of a full bomb.

#### Describe the solution
Add 4 new items:
The magnetron/microwave generator, a component of EMP grenades, that can be found in microwaves.
The flux compression generator, a component of EMP grenades, that can be rarely found in labs, or crafted.
The large copper tube, a component of the flux compression generator, that can be forged.
The EMP bomb, an explosive device that puts out an EMP pulse. This replaces the emp grenade.

I also moved an equivalent of the EMP grenade to be find/buy from Hub 01 only.
I also obsoleted the recipe for the scrambler grenade, and it can only be found or bought from Hub 01.

See 2fb335d7f202e57b9d364916a3f092f88abccefb for more detailed commentary.

#### Describe alternatives you've considered
There may be some other equivalents for the capacitor bank, but they need some special handling.

#### Testing
Tried crafting everything.
empblast
empuse

#### Additional context
This EMP device works by draining a high-current capacitor (represented
by a plutonium fuel cell) into a 'explosively propelled flux compression
generator'. When this generator is activated, with the aid of an
explosion, it creates a strong magnetic field. This magnetic field is
then passed through a microwave generator (magnetron), into a single
loop antenna, which creates the EMP.

The items added could spawn in more places.

TODO:
- [x] insulators
- [x] Fudge explosion less

### CODE REVIEW

**Comment by esotericist** [comment]
The EMP bomb should probably have more (non-ferromagnetic) materials in its construction to account for the insulation needed to keep it from cooking itself before it goes off. See page 5 (excluding cover sheet).

**Comment by Fris0uman** [comment]
Should some EMP bombs spawn in labs baracks, and other military installations?

**Comment by Funguss** [comment]
No more scramblers without trading? Daaaamn... Gonna miss saving those cyborgs in lab starts. I don't like it but I can't argue, hehe.
What's the case for introducing a large copper tube? Why not use the current copper pipe?

**Comment by Rail-Runner** [comment]
> I also obsoleted the recipe for the scrambler grenade, and it can only be found or bought from Hub 01.

Well, for me it doesn't make sense in current form at all, even if uncraftable. Based on what it does, it seems to be just a radio transmitter of some kind; those are not one-use usually, and I guess one could build such a transmitter from salvaged robot parts related to sending IFF signals to other robots, if they're famillar with how IFF signals work.

**Comment by anothersimulacrum** [comment]
@Funguss Smaller pipe isn't large enough. I'm making size adjustments though, and we'll see where we end up.

@Rail-Runner The only reason I touched the scrambler is because it's made from EMP grenades. If someone wants to make more extensive changes, they can, but I didn't really want to.

**Comment by AlexanderBodnya** [comment]
This effectively removes one of the legit ways to get out from the lab. The game is striving for realism, I get it, but from game design standpoint it just takes yet another element of gameplay from the player. You may argue, that it makes emp grenades more valuable, and hunting them and hub01 adds to the game and it may, but please add all parts of the bomb to the lab drop, otherwise you are making not only positive, but also a negative change.

**Comment by kevingranade** [comment]
If a balance change makes lab starts not work, we're more likely to remove lab starts than to work around it with arbitrary patches. Lan starts ste not a core feature.

File: data/json/recipes/electronics/parts.json [hunk 1]
```diff
@@ -21,6 +21,24 @@
...
+    "difficulty": 6,
+    "skills_required": [ "fabrication", 4 ],
+    "time": "3h 20m",
+    "reversible": true,
+    "book_learn": [ [ "manual_electronics", 6 ] ],
+    "using": [ [ "soldering_standard", 10 ], [ "surface_heat", 50 ], [ "welding_standard", 5 ] ],
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "tools": [ [ [ "mold_plastic", -1 ] ] ],
+    "//": "Fill pipe with RDX, wind copper tubes into coil around it, leaving empty space. Use plastic (representing fiberglass) to insulate wires and supporting the coil of wires",
+    "//1": "plastic matches fiberglass required roughly by volume (4804mL), not by weight",
```
Comment (ymber) [inline data/json/recipes/electronics/parts.json]: Fiberglass and common polymers are completely different. All of the materials the paper describes for this use would be hard to work with for a survivor and we don't have most of them in the game. I don't think this should be craftable unless we add a load of other crafting stuff for working with those materials.

File: data/json/items/electronics.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "material": "steel",
+    "symbol": ":",
+    "color": "light_gray"
+  },
+  {
+    "id": "flux_comp_gen",
+    "type": "GENERIC",
+    "category": "spare_parts",
+    "name": "explosively pump flux compression generator",
+    "description": "This large device consists mainly of a tube of copper wire surrounding a large copper tube filled with high explosives.  When detonated properly, the explosives allow the device to large amounts of electrical energy in a very short time.",
```
Comment (ymber) [inline data/json/items/electronics.json]: Missing word near the end of the description.

File: data/json/recipes/electronics/parts.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/recipes/electronics/parts.json]: Okay, removed the recipe.

File: data/json/items/electronics.json (hunk 2 above)
Comment (anothersimulacrum) [reply data/json/items/electronics.json]: Yep, thank you!

**Comment by anothersimulacrum** [comment]
Changes have been made (side note: I'm really bad at keeping my cursor out of the image)

lab-clair
lab-use

**Comment by ave-** [comment]
So, people with npcs disabled are essentially locked out of scrambler grenades entirely with this PR?

**Comment by anothersimulacrum** [comment]
Yes.

**Comment by ave-** [comment]
@kevingranade is this acceptable? I do not know what is the current stance on players with npc's disabled.

**Comment by Zireael07** [comment]
Wait a moment, scrambler grenades no longer spawn? I thought it was just about making them not craftable?

**Comment by anothersimulacrum** [comment]
I didn't adjust spawns, they never spawned before.

**Comment by ave-** [comment]
They were craftable-only for a long time (may be since implemented, but I'm not sure). That's why I'm worried about they disappearance for the 'lone wolf' types of people like me (I like to play with npcs disabled as even in a current quite good status I don't think they are worth to enable).

As a solution, can you add the scrambler grenade to the same spawnable itemgroups as the new emp grenade, but with like 10% chance relative to the emp one?

**Comment by Zireael07** [comment]
What are the usual spawn places for high-tech stuff? Once upon a time, bank safes were one, and other than that, IIRC labs?

**Comment by I-am-Erk** [comment]
We're not just going to keep balancing the game around every hypothetical niche playstyle. Trading and getting quests from static NPCs is pretty basic stuff and has been the default setting for a long time now.

**Comment by ave-** [comment]
Move an option to disable NPC's from options to the mod then? Like it is done with vitamins, npc needs and other No_X mods. From what you saying it looks like they are pretty much playable now and with PR's like this they are **required** to have access to all vanilla items.

**Comment by kevingranade** [comment]
We do not make things craftable or not based on game balance, that is strictly based on whether we think it's feasible for a lone survivor to make them.

An option to toggle recipes based on the presence of NPCs is not ok since it contradicts our design principles. This would need to be a mod.

Frankly I'm not even clear on what a "scrambler grenade" is supposed to be.

**Comment by ave-** [comment]
'Scramber' is something that disrupts IFF. Current real implementations can disable IFF (ie all targets considered enemies), but not the reverse.

To make all turret targets 'friendly' you need to precisely know the IFF protocol and overwhelm the IFF frequencies with successful identifications. Ie not realistic, but realistic conception (noise IFF freqs so much that noone can identify so everyone considered enemies) is completely unusable for a player because in most cases turrets are strategically placed where there are no targets except the player.

**Comment by I-am-Erk** [comment]
Why would the option be moved to a mod? There are many ways you can change options in the game to make things more difficult. These include turning off NPCs, increasing monster speed, and decreasing item spawns. We don't do any of our development based around the assumption that people are going to change the options to various different combinations. The game is perfectly playable without scrambler grenades, and they're but one of an increasing number of things that will be harder if you're trying to do a playthrough as a sole survivor in a dead world.

**Comment by ave-** [comment]
I think the simplest solution would be moving the recipe to an appropriate mod. If scrambler grenade in it's current form is not entirely realistic then may be Aftershock mod would fit?

**Comment by I-am-Erk** [comment]
Sure, the recipe for it could be put into aftershock. I don't think any of us care too much about stuff like that

**Comment by esotericist** [comment]
Personally, I think the device itself is also aftershock material. Its theory of operation is kinda shaky.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/emp-grenades-how-acquire-recipe/21182/11

**Pull request merged**